src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs

```diff
@@ -28,6 +28,75 @@ public JsonConsoleFormatter(IOptionsMonitor<JsonConsoleFormatterOptions> options
             _optionsReloadToken = options.OnChange(ReloadLoggerOptions);
         }
 
+        private static List<Exception> GetInnerExceptions(Exception exception)
+        {
+            List<Exception> result = new List<Exception>();
+            if (exception.InnerException != null)
+            {
+                result.Add(exception.InnerException);
+            }
+            AggregateException aggregated = exception as AggregateException;
+            if (aggregated != null)
+            {
+                foreach (var inner in aggregated.InnerExceptions)
+                {
+                    if (inner != exception.InnerException)
+                    {
+                        result.Add(inner);
+                    }
+                }
+            }
+            return result;
+        }
+
+        private static void WriteException(Utf8JsonWriter writer, Exception exception, int maxDepth, bool includePropertyName)
+        {
+            if (includePropertyName)
+            {
+                writer.WriteStartObject(nameof(Exception));
+            }
+            else
+            {
+                writer.WriteStartObject();
+            }
+            writer.WriteString(nameof(exception.Message), exception.Message);
+            writer.WriteString("Type", exception.GetType().ToString());
+
+            writer.WriteStartArray(nameof(exception.StackTrace));
+            {
+                string stackTrace = exception?.StackTrace;
+                if (stackTrace != null)
+                {
+#if NETCOREAPP
+                    foreach (var stackTraceLines in stackTrace?.Split(Environment.NewLine))
+#else
+                            foreach (var stackTraceLines in stackTrace?.Split(new string[] { Environment.NewLine }, StringSplitOptions.None))
+#endif
+                    {
+                        writer.WriteStringValue(stackTraceLines);
+                    }
+                }
+            }
+            writer.WriteEndArray();
+            writer.WriteNumber(nameof(exception.HResult), exception.HResult);
+            if (maxDepth > 0)
```

Comment:
You could perhaps add a test that exercises case when maxDepth has negative value.